### ISSUE TITLE
'Cannot read property of undefined' when running with 'project: tsconfig.json'(eslintrc) and 'module: none' (tsconfig) 

### ISSUE THREAD

**Issue by undeadcat** [typescript-eslint/typescript-eslint#497]
**What code were you trying to parse?**
ts.ts
```ts
declare var React;
declare var PropTypes;
const SHOULD_BE_UNUSED = 1;

const INPUT_PASS_PROPS = {
    autoFocus: true,
    disabled: true,
    warning: true,
    error: true,
    size: true,
    onKeyDown: true,
};

console.log(PropTypes.number);
console.log(React.Component);
console.log(INPUT_PASS_PROPS);
```
.eslintrc.yaml:
```yaml
parser: "@typescript-eslint/parser"
root: true
plugins:
  - "@typescript-eslint"

parserOptions:
  project: "./tsconfig.json"
rules:
  semi: [error, always]
  no-extra-semi: error
  "@typescript-eslint/no-unused-vars": error
```
tsconfig.json
```
{
  "compilerOptions": {
    "module": "none",
    "target": "es5"
  },
  "exclude": [
    "node_modules"
  ]
}
```
To reproduce, save the above files in a directory and invoke `node_modules/.bin/eslint ts.ts`.

**What did you expect to happen?**
Expected errors to be shown, the code contains an unused var that should have been reported by the no_unused_vars rule.
**What actually happened?**
An error: `error  Parsing error: Cannot read property 'text' of undefined`.
No rule violations are reported. 
ESLint loses the stack, even with --debug, but breaking at https://github.com/eslint/eslint/blob/master/lib/linter.js#L524 allows to see the stack in the debugger:
```stacktrace
TypeError: Cannot read property 'text' of undefined
    at TokenObject.TokenOrIdentifierObject.getText (/Users/jetbrains/sources/testProjects/typescript-eslint-bug/node_modules/typescript/lib/typescript.js:120347:31)
    at Converter.convertNode (/Users/jetbrains/sources/testProjects/typescript-eslint-bug/node_modules/@typescript-eslint/typescript-estree/dist/convert.js:1414:31)
    at Converter.converter (/Users/jetbrains/sources/testProjects/typescript-eslint-bug/node_modules/@typescript-eslint/typescript-estree/dist/convert.js:66:27)
    at Converter.convertChild (/Users/jetbrains/sources/testProjects/typescript-eslint-bug/node_modules/@typescript-eslint/typescript-estree/dist/convert.js:138:21)
    at Converter.convertNode (/Users/jetbrains/sources/testProjects/typescript-eslint-bug/node_modules/@typescript-eslint/typescript-estree/dist/convert.js:601:32)
    at Converter.converter (/Users/jetbrains/sources/testProjects/typescript-eslint-bug/node_modules/@typescript-eslint/typescript-estree/dist/convert.js:66:27)
    at Converter.convertChild (/Users/jetbrains/sources/testProjects/typescript-eslint-bug/node_modules/@typescript-eslint/typescript-estree/dist/convert.js:138:21)
    at result.createNode.declarations.node.declarationList.declarations.map.el (/Users/jetbrains/sources/testProjects/typescript-eslint-bug/node_modules/@typescript-eslint/typescript-estree/dist/convert.js:615:84)
    at Array.map (<anonymous>)
    at Converter.convertNode (/Users/jetbrains/sources/testProjects/typescript-eslint-bug/node_modules/@typescript-eslint/typescript-estree/dist/convert.js:615:69)

```
**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `1.7.1-alpha.20` |
| `@typescript-eslint/eslint-plugin` | `1.7.1-alpha.20` |
| `TypeScript`                | `3.3.4000` |
| `ESLint`                    | `5.15.3` |
| `node`                      | `v10.5.0` |
| `npm`                       | `6.4.1` |

**Comment by bradzacher** [comment]
I'm unable to reproduce this when parsing against master.

Similarly it works fine on astexplorer: https://astexplorer.net/#/gist/07b6d48a61c07b10729d0577de1b2875/a56abd3bf1c62657bc77dd012a06faf332700efe

**Comment by bradzacher** [comment]
I'll close this in favour of #498
